cashu/mint/ledger.py

```diff
@@ -177,7 +177,11 @@ async def _generate_promise(
         C_ = b_dhke.step2_bob(B_, private_key_amount)
         logger.trace(f"crud: _generate_promise storing promise for {amount}")
         await self.crud.store_promise(
-            amount=amount, B_=B_.serialize().hex(), C_=C_.serialize().hex(), db=self.db
+            amount=amount,
+            B_=B_.serialize().hex(),
+            C_=C_.serialize().hex(),
+            id=keyset.id,

```

Comment:
we now store the "id" for a promise in the wallet db because we need it for restore